### PR TITLE
model api improvements

### PR DESCRIPTION
**Pull request by akorobeinikov**
`query_state` don't relate to status of request. Now all exceptions raised in callback will be move to `callback_exceptions` list.
As I know `wait_all` for `AsyncQueue`  guarantees that callback is finished. @AlexeyLebedev1 Am I right?

### CODE REVIEW

**Review by vladimir-dudnik — APPROVED** [review]

File: models/public/device_support.md [hunk 1]
```diff
@@ -4,106 +4,106 @@
...
-| densenet-121-tf | YES |  | YES |
+| densenet-121-tf | YES | YES | YES |
 | detr-resnet50 | YES |    |    |
 | dla-34 | YES | YES | YES |
-| drn-d-38 | YES |  |    |
-| efficientdet-d0-tf | YES |  | YES |
-| efficientdet-d1-tf | YES |  | YES |
+| drn-d-38 | YES | YES |    |
+| efficientdet-d0-tf | YES | YES | YES |
+| efficientdet-d1-tf | YES | YES | YES |
```
Comment (ivikhrev) [inline models/public/device_support.md:28]:
```suggestion
| efficientdet-d0-tf | YES |    | YES |
| efficientdet-d1-tf | YES |    | YES |
```

File: models/public/device_support.md [hunk 2]
```diff
@@ -4,106 +4,106 @@
...
+| googlenet-v3 | YES | YES | YES |
+| googlenet-v3-pytorch | YES | YES | YES |
 | googlenet-v4-tf | YES | YES | YES |
 | gpt-2 | YES | YES | |
 | hbonet-0.25 | YES | YES | YES |
 | hbonet-1.0 | YES | YES | YES |
-| higher-hrnet-w32-human-pose-estimation | YES |  |    |
-| hrnet-v2-c1-segmentation | YES |  |    |
+| higher-hrnet-w32-human-pose-estimation | YES | YES |    |
+| hrnet-v2-c1-segmentation | YES | YES |    |
```
Comment (ivikhrev) [inline models/public/device_support.md:59]:
```suggestion
| hrnet-v2-c1-segmentation | YES |    |    |
```

File: models/public/device_support.md [hunk 3]
```diff
@@ -4,106 +4,106 @@
...
+| mobilenet-yolo-v4-syg | YES | YES |    |
 | mozilla-deepspeech-0.6.1 | YES |    |    |
 | mozilla-deepspeech-0.8.2 | YES |    |    |
 | mtcnn-o | YES | YES |    |
 | mtcnn-p | YES | YES |    |
 | mtcnn-r | YES | YES |    |
 | netvlad-tf | YES |    |    |
 | nfnet-f0 | YES | YES | YES |
-| ocrnet-hrnet-w48-paddle | YES |  | |
+| ocrnet-hrnet-w48-paddle | YES | YES | |
```
Comment (ivikhrev) [inline models/public/device_support.md:88]:
```suggestion
| ocrnet-hrnet-w48-paddle | YES |    |    |
```

**Review by ivikhrev — COMMENTED** [review]
These models failed with stak overflow issue, not that one caused by `query_state()`.

File: models/intel/device_support.md [hunk 4]
```diff
@@ -17,18 +17,18 @@
 | driver-action-recognition-adas-0002-decoder | YES | YES | YES |
 | driver-action-recognition-adas-0002-encoder | YES | YES | YES |
 | emotions-recognition-retail-0003 | YES | YES | YES |
-| face-detection-0200 | YES |  | YES |
-| face-detection-0202 | YES |  | YES |
-| face-detection-0204 | YES |  | YES |
+| face-detection-0200 | YES | YES | YES |
+| face-detection-0202 | YES | YES | YES |
+| face-detection-0204 | YES | YES | YES |
 | face-detection-0205 | YES |  | YES |
```
Comment (vladimir-dudnik) [inline models/intel/device_support.md:23]: 205/206 is not the same as 200..204?

File: models/intel/device_support.md [hunk 5]
```diff
@@ -64,15 +64,15 @@
...
 | person-attributes-recognition-crossroad-0238 | YES | YES | YES |
 | person-detection-0106 | YES |     |     |
-| person-detection-0200 | YES |  | YES |
-| person-detection-0201 | YES |  | YES |
-| person-detection-0202 | YES |  | YES |
+| person-detection-0200 | YES | YES | YES |
+| person-detection-0201 | YES | YES | YES |
+| person-detection-0202 | YES | YES | YES |
 | person-detection-0203 | YES | YES | YES |
 | person-detection-0301 | YES |  |  |
```
Comment (vladimir-dudnik) [inline models/intel/device_support.md:77]: 0301/0302 not about state?

File: models/public/device_support.md [hunk 6]
```diff
@@ -4,106 +4,106 @@
...
+| densenet-121-tf | YES | YES | YES |
 | detr-resnet50 | YES |    |    |
 | dla-34 | YES | YES | YES |
-| drn-d-38 | YES |  |    |
-| efficientdet-d0-tf | YES |  | YES |
-| efficientdet-d1-tf | YES |  | YES |
+| drn-d-38 | YES | YES |    |
+| efficientdet-d0-tf | YES |    | YES |
+| efficientdet-d1-tf | YES |    | YES |
 | efficientnet-b0 | YES |  | YES |
```
Comment (vladimir-dudnik) [inline models/public/device_support.md:29]: this one ?

Comment (ivikhrev) [reply models/public/device_support.md:29]: efficientdet-d0-tf  & efficientdet-d1-tf failed with stack overflow, is it already solved?

Comment (ivikhrev) [reply models/public/device_support.md:29]: > efficientdet not enabled on GPU by this change

it was, and I thought your question was about that, ok.

File: models/intel/device_support.md (hunk 4 above)
Comment (akorobeinikov) [reply models/intel/device_support.md:23]: I don't know why 205/206 was GPU off in [PR](https://github.com/openvinotoolkit/open_model_zoo/pull/3261). We can add and try

Comment (vladimir-dudnik) [reply models/intel/device_support.md]: I missed them in my big PR and then gad to exclude one by one missed models

File: models/public/device_support.md (hunk 6 above)
Comment (akorobeinikov) [reply models/public/device_support.md:29]: I think `efficientnet-b0` is ok and was GPU off by mistake, but we can't test this model as I see. (there is not test case for it)

**Pull request merged**